### PR TITLE
WAR for supporting the rotation + residual pattern

### PR DESCRIPTION
**Pull request by naoyam**
Stacked on top of #3549. This is also a WAR for #3455 and necessary to schedule RoPE-like rotation patterns. 

Because of the issue, a tensor may have two IDs that are exactly mapped. For example, when an ID is sliced to half and then is padded back to the same size, and the final output ID is used with the initial input ID, the initial input and the final output IDs get mapped together. This can make it difficult to use `scheduleLoopDomainsLike`. For example, if a reference has a split that is done with the final output ID, and we want to replay the split on other tensors, it becomes ambiguous whether the split is done with the initial input or the final output since both are exactly mapped.

To avoid this ambiguity, this PR adds a flag to indicate that we just want to update the current loop domain with a reference domain. As seen in the added tests, this flag is used to propagate the scheduling of a reference tensor once all resize ops are propagated to inputs. Specifically, the overall scheduling follows this pattern:

1. Propagate all slice and pad ops to fusion inputs
2. Pick and schedule a reference tensor
3. Propagate the scheduling of the reference tensor to the other tensors

`scheduleLoopDomainsLike` with the flag is used at step 3. For that step, we know that we don't need to schedule each tensor with a complex replay path, like some backward ops followed by some other forward ops, but we just need to update the current loop domain by replaying the diff with the reference domain.

### CODE REVIEW

**Comment by naoyam** [comment]
!test

**Comment by naoyam** [comment]
!build

File: csrc/scheduler/tools/loop_domain_scheduler.h [hunk 1]
```diff
@@ -22,7 +22,8 @@ namespace scheduler_tools {
 // each tensor by replaying exprs found in the Exact ValGraph.
 void scheduleLoopDomainsLike(
     const std::vector<TensorView*>& tvs,
-    const std::vector<IterDomain*>& ref_loop_dom);
+    const std::vector<IterDomain*>& ref_loop_dom,
+    bool update_mode = false);
```
Comment (jacobhinkle) [inline csrc/scheduler/tools/loop_domain_scheduler.h:26]: We should document `update_mode` in the comment here.

File: csrc/scheduler/tools/loop_domain_scheduler.cpp [hunk 2]
```diff
@@ -188,9 +193,13 @@ void LoopDomainScheduler::schedule(TensorView* tv) const {
 
   // All of the existing IDs are reused as much as possible to
   // minimize creating new IDs.
-  auto all_ids = tv->domain()->allIDs();
+
   std::unordered_map<ValGroup, IterDomain*> group_to_id;
   ValGroups all_id_groups;
+  // When update_mode_ is true, only the loop domain IDs are reused as
+  // we attemp to transform the current loop domain to look like the
```
Comment (jacobhinkle) [inline csrc/scheduler/tools/loop_domain_scheduler.cpp:200]:
```suggestion
  // we attempt to transform the current loop domain to look like the
```

File: csrc/scheduler/tools/loop_domain_scheduler.h (hunk 1 above)
Comment (jacobhinkle) [reply csrc/scheduler/tools/loop_domain_scheduler.h:26]: Would a better name be `update_loop_domain_only`? That name implies that we will not be modifying the logical domain to place the loop as a producer of logical.

**Review by jacobhinkle — APPROVED** [review]
LGTM. Only comment is minor about naming of the param `update_mode`.

**Comment by naoyam** [comment]
!build

**Pull request merged**